lib/ditcher.js

```diff
@@ -45,13 +46,83 @@ var Ditcher = function (cfg, lgr, versionNumber, callback) {
 
 util.inherits(Ditcher, EventEmitter);
 
+/**
+ * Creates an object containing the expression key and value queried.
+ *
+ * @param {String} expression - This is the operator used in query.
+ * @param {ObjectId|Number|Boolean} value - This is the value being quried.
+ *
+ * @returns {Object} expressionQuery - contains the expressions key and query value.
+ */
+function expressionQuery(expression, value) {
+  var expressionQuery = {};
+  expressionQuery[expression] = value;
+  return expressionQuery[expression];
+}
+
+// contains functions to convert the query value based on the query type.
+var convertors = {
+  "Date": function (value) {
+    return new Date(value);
+  },
+  "ObjectId": function (value) {
+    return ObjectId(value);
+  },
+  "Any": function (value, field, expression) {
+    var arr = [];
+    var query = {};
+    var str = value;
+    query[field] = str;
+    arr.push(query);
+    try {
+      query[field] = expression ? expressionQuery(expression, value) : ObjectId(value);
+      arr.push(query);
+    } catch (err) { }
+    var num = Number(value);
+    if (!Number.isNaN(num)) {
+      query[field] = expression ? expressionQuery(expression, num) : num;
+      arr.push(query);
+    }
+
+    if (value.toLowerCase() === 'true') {
+      query[field] = expression ? expressionQuery(expression, true) : true;
+      arr.push(query);
+    } else if (value.toLowerCase() === 'false') {
+      query[field] = expression ? expressionQuery(expression, false) : false;
+      arr.push(query);
+    }
+    return arr;
+  }
+}
+
+/**
+ * Converts the query based on the type of query and if an expression was used
+ *
+ * @param {Object} type - Type will contain the type and value being queried. Type can have these values: "String", "Number", "Boolean", "Date", "ObjectId" or "Any".
+ * @param {String} field - This is the column being quried.
+ * @param {String} expression - This is the operator used in query e.g. "$ne" (not equal).
+ *
+ * @returns {String|Number|Boolean|Object} 
+ */
+function convertQueryType(type, field, expression) {
+  if (convertors[type.type] === undefined) {
+    return type.value;
+  } else {
+    return convertors[type.type](type.value, field, expression);
+  }
+}
+
 var crit_ops = {
   eq: function (query, fields) {
     var field;
     if (null !== fields) {
-      for (field in  fields) {
+      for (field in fields) {
         if (fields.hasOwnProperty(field)) {
-          query[field] = fields[field];
+          if (fields[field].type === "Any") {
+            query['$or'] = convertQueryType(fields[field], field, undefined);
```

Comment:
But if you want to be explicit about the fact that there is another parameter but it should be undefined here then it's ok.